pytorch_lightning/trainer/trainer.py

```diff
@@ -1702,9 +1703,10 @@ def world_size(self) -> int:
     @property
     def should_rank_save_checkpoint(self) -> bool:
         rank_zero_deprecation(
-            "`Trainer.should_rank_save_checkpoint` is deprecated in v1.6 and will be removed in 1.8.", stacklevel=5
+            "`Trainer.should_rank_save_checkpoint` is deprecated in v1.6 and will be removed in v1.8.", stacklevel=5
         )
-        return self.training_type_plugin.should_rank_save_checkpoint
+        ttp = self.training_type_plugin
+        return isinstance(ttp, TPUSpawnPlugin) and ttp.local_rank == 0 or ttp.is_global_zero
```

Comment:
re: custom plugins like @awaelchli 's comment [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/11068#pullrequestreview-833560843):
could check for `hasattr(should_rank_save_checkpoint)` but could be misleading when TTP's `should_rank_save_checkpoint` is removed and doesn't actually control anything but the trainer property is not yet removed

Reply:
I'd say this is okay since it is read-only. IMO the deprecation here is more about not explicitly breaking the property access.

I also wouldn't worry about custom plugins since we are absolutely not recommending them for now. And when we do (1.6+), users won't implement `should_rank_save_checkpoint` as this PR will have removed it.

Reply:
I still think if we are not going all the way, we should remove the Trainer property without deprecation, or raise a custom error message. After all, it was just meant to be a short hand for `trainer.training_type_plugin.xyz`

Reply:
I'm totally fine either way. I proposed marking the property protected from the start
https://github.com/PyTorchLightning/pytorch-lightning/pull/7684#discussion_r639336673

Reply:
Keeping the deprecation since nobody seems to have a strong opinion and the work has been done already